src/bin/cargo/commands/add.rs

```diff
@@ -242,7 +242,24 @@ fn parse_dependencies(config: &Config, matches: &ArgMatches) -> CargoResult<Vec<
         .flatten()
         .map(|c| (Some(c.clone()), None))
         .collect::<IndexMap<_, _>>();
+
     let mut infer_crate_name = false;
+
+    for krate in crates.iter() {
+        let crate_name = krate.0.clone();
+        if let Some(crate_name) = crate_name {
+            if !crate_name.starts_with("+") {
+                continue;
+            }
+
+            let crate_name_wo_prefix = crate_name[1..].to_string();
```

Comment:
Unnecessary `to_string` allocation.

```suggestion
            let crate_name_wo_prefix = &crate_name[1..];
```